File: src/Envirotrax.App/Envirotrax.App.Server/Data/Models/Professionals/Insurances/ProfessionalInsurance.cs

using System.ComponentModel.DataAnnotations;
using Envirotrax.App.Server.Data.Models.Users;
using Envirotrax.Common.Data.Attributes;
using Envirotrax.Common.Data.Models;

namespace Envirotrax.App.Server.Data.Models.Professionals.Insurances;

public class ProfessionalInsurance : IProfessionalModel, ICreateAuditableModel<AppUser>
{
    [AppPrimaryKey(true)]
    public int Id { get; set; }

    public int ProfessionalId { get; set; }
    public Professional? Professional { get; set; }

    public DateTime? ExpirationDate { get; set; }

    [Required]
    [StringLength(50)]
    public string InsuranceNumber { get; set; } = null!;

    [Required]
    public string FilePath { get; set; } = null!;

    public int? CreatedById { get; set; }
    public AppUser? CreatedBy { get; set; }
    public DateTime CreatedTime { get; set; }
}

File: src/Envirotrax.Admin/envirotrax.admin.client/src/app/shared/models/professionals/professional-insurance.ts
import { ReferencedProfessional } from './professional';
import { ExpirationType } from './licenses/professional-user-license';

export interface ProfessionalInsurance {
    id?: number;
    professional?: ReferencedProfessional;
    expirationDate?: string;
    insuranceNumber?: string;
    filePath?: string;
    expirationType?: ExpirationType;
}
